examples/w3c_tracecontext_test/main.cc

```diff
@@ -0,0 +1,187 @@
+#include "opentelemetry/context/runtime_context.h"
+#include "opentelemetry/exporters/ostream/span_exporter.h"
+#include "opentelemetry/ext/http/client/curl/http_client_curl.h"
+#include "opentelemetry/ext/http/server/http_server.h"
+#include "opentelemetry/sdk/trace/simple_processor.h"
+#include "opentelemetry/sdk/trace/tracer_provider.h"
+#include "opentelemetry/trace/propagation/http_trace_context.h"
+#include "opentelemetry/trace/provider.h"
+#include "opentelemetry/trace/scope.h"
+
+#include <algorithm>
+#include "nlohmann/json.hpp"
+
+namespace
+{
+static opentelemetry::trace::propagation::HttpTraceContext<std::map<std::string, std::string>>
+    propagator_format;
+
+void Setter(std::map<std::string, std::string> &carrier,
+            nostd::string_view trace_type        = "traceparent",
+            nostd::string_view trace_description = "")
+{
+  carrier[std::string(trace_type)] = std::string(trace_description);
+}
+
+nostd::string_view Getter(const std::map<std::string, std::string> &carrier,
+                          nostd::string_view trace_type = "traceparent")
+{
+  auto it = carrier.find(std::string(trace_type));
+  if (it != carrier.end())
+  {
+    return nostd::string_view(it->second);
+  }
+  return "";
+}
+
+void initTracer()
+{
+  auto exporter = std::unique_ptr<sdktrace::SpanExporter>(
+      new opentelemetry::exporter::trace::OStreamSpanExporter);
+  auto processor = std::shared_ptr<sdktrace::SpanProcessor>(
+      new sdktrace::SimpleSpanProcessor(std::move(exporter)));
+  auto provider = nostd::shared_ptr<opentelemetry::trace::TracerProvider>(
+      new sdktrace::TracerProvider(processor));
+  // Set the global trace provider
+  opentelemetry::trace::Provider::SetTracerProvider(provider);
+}
+
+nostd::shared_ptr<opentelemetry::trace::Tracer> get_tracer()
+{
+  auto provider = opentelemetry::trace::Provider::GetTracerProvider();
+  return provider->GetTracer("w3c_tracecontext_test");
+}
+
+struct Uri
+{
+  std::string host;
+  uint16_t port;
+  std::string path;
+
+  Uri(std::string uri)
+  {
+    size_t host_end = uri.substr(7, std::string::npos).find(":");
+    size_t port_end = uri.substr(host_end + 1, std::string::npos).find("/");
+
+    host = uri.substr(0, host_end + 7);
+    port = std::stoi(uri.substr(7 + host_end + 1, port_end));
+    path = uri.substr(host_end + port_end + 2, std::string::npos);
+  }
+};
```

Comment:
jfi - i added a url_parser.h header to parse the url as part of zipkin exporter PR  ( https://github.com/open-telemetry/opentelemetry-cpp/pull/471/files )

Reply:
I'd love to use that once it's merged.